### PR TITLE
chore: remove BooleanFieldValue

### PR DESCRIPTION
**Pull request by jelbourn**
Based on top of #1289 

This removes the custom decorator as it prevents tree-shaking of components that use it. Replaced with a `coerceBooleanProperty` function.

### CODE REVIEW

File: src/lib/button-toggle/button-toggle.ts [hunk 1]
```diff
@@ -21,7 +21,8 @@ import {
     FormsModule,
 } from '@angular/forms';
 import {Observable} from 'rxjs/Observable';
-import {BooleanFieldValue, MdUniqueSelectionDispatcher} from '@angular2-material/core';
+import {MdUniqueSelectionDispatcher} from '@angular2-material/core';
+import {coerceBooleanProperty} from '../core/coersion/boolean-property';
```
Comment (hansl) [inline src/lib/button-toggle/button-toggle.ts]: Either wait for my PR to be merged in (where you can use `../core`) or fix those imports to pass the linting.

**Comment by jelbourn** [comment]
Yeah, I planned on waiting for your PR

**Comment by jelbourn** [comment]
@hansl rebased

File: src/lib/slide-toggle/slide-toggle.ts [hunk 2]
```diff
@@ -65,13 +60,18 @@ export class MdSlideToggle implements AfterContentInit, ControlValueAccessor {
...
   private _slideRenderer: SlideToggleRenderer = null;
 
-  @Input() @BooleanFieldValue() disabled: boolean = false;
   @Input() name: string = null;
   @Input() id: string = this._uniqueId;
   @Input() tabIndex: number = 0;
   @Input() ariaLabel: string = null;
   @Input() ariaLabelledby: string = null;
 
+  private _disabled: boolean = false;
```
Comment (DevVersion) [inline src/lib/slide-toggle/slide-toggle.ts]: Can we move this up to the other private variables? And also make the `_hasFocus` private while you are at it? :)

**Comment by jelbourn** [comment]
@kara rebased and ready for review

**Pull request merged**